src/index.ts

```
@@ -1,29 +1,37 @@
-import fetch from 'cross-fetch';
+import { ofetch as fetch } from 'ofetch';
 
 const PINEAPPLE_URL = 'https://pineapple.fyi';
+const timeout = 10e3;
 
-export async function pin(json: any, url: string = PINEAPPLE_URL) {
+export async function pin(json: any, url = PINEAPPLE_URL) {
   const init = {
     method: 'POST',
     headers: {
       Accept: 'application/json',
       'Content-Type': 'application/json'
     },
-    body: JSON.stringify({
+    body: {
       jsonrpc: '2.0',
       method: 'pin',
       params: json,
       id: null
-    })
+    },
+    timeout
   };
-  const res = await fetch(url, init);
-  const content = await res.json();
-  return content.result || { error: content.error };
+
+  return sendRequest(url, init);
 }
 
 export async function upload(body: any, url = `${PINEAPPLE_URL}/upload`) {
-  const init = { method: 'POST', body };
-  const res = await fetch(url, init);
-  const content = await res.json();
-  return content.result || { error: content.error };
+  const init = { method: 'POST', body, timeout };
+
+  return sendRequest(url, init);
+}
+
+async function sendRequest(url: string, init: any) {
+  try {
+    return (await fetch(url, init)).result;
+  } catch (e: any) {
+    return { error: e.data?.error || { code: e.status, message: e.statusText } };
```

it should throw the error due to usage [here](https://github.com/snapshot-labs/snapshot/blob/94d82953023f9e90a8a1027490711db2b71736ea/src/components/InputUploadImage.vue#L25), [here](https://github.com/snapshot-labs/snapshot-sequencer/blob/01b5c1ad78f56704e17eace3bc2e56f101442a0c/src/ingestor.ts#L195) and [here](https://github.com/snapshot-labs/sx-ui/blob/8412e229e838cc45684f4b81d629f26fa758a2ef/src/helpers/utils.ts#L345)

---

[exception](https://github.com/snapshot-labs/snapshot/blob/94d82953023f9e90a8a1027490711db2b71736ea/src/composables/useImageUpload.ts#L44)

---

Updated to return a `Promise.rejection` on error. This will handle all the cases where we use try/catch when using `pin`

---

For the exception case (https://github.com/snapshot-labs/snapshot/blob/94d82953023f9e90a8a1027490711db2b71736ea/src/composables/useImageUpload.ts#L44), it'll need some refactoring, as the new implementation will always return a `Promise.rejection` on error, and always in the form of a JSON-RPC object.